params/config.go

```diff
@@ -368,6 +386,15 @@ func (c *ChainConfig) IsKIP103ForkBlock(num *big.Int) bool {
 	return c.Kip103CompatibleBlock.Cmp(num) == 0
 }
 
+// IsRandaoForkBlockParent returns whethere num is one block before the randao block.
+func (c *ChainConfig) IsRandaoForkBlockParent(num *big.Int) bool {
+	if c.RandaoCompatibleBlock == nil || num == nil {
+		return false
+	}
+	nextNum := new(big.Int).Add(num, common.Big1)
+	return c.RandaoCompatibleBlock.Cmp(nextNum) == 0 // randao == num + 1
```

Comment:
How can we handle the case: `RandaoCompatibleBlock == 0`?

Reply:
In that case, `IsRandaoForkBlockParent` returns false.
You'd have to use `AllocRegistry` to generate the GenesisAlloc with the initial Registry state.